pkg/controller/replication/replication_controller.go

```diff
@@ -632,5 +634,66 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 		updatedRC.Status.AvailableReplicas != *(updatedRC.Spec.Replicas) {
 		rm.enqueueControllerAfter(updatedRC, time.Duration(updatedRC.Spec.MinReadySeconds)*time.Second)
 	}
+
+	pods, err := rm.podLister.Pods(rc.Namespace).List(labels.Set(rc.Spec.Selector).AsSelectorPreValidated())
+	if err != nil {
+		return err
+	}
+	// Find rejected pods and delete them.
+	for _, pod := range pods {
+		if controller.IsPodRejected(pod) {
+			// Ignore deletion error here.
+			rm.podControl.DeletePod(pod.Namespace, pod.Name, rc)
+		}
+	}
+
+	boolPtr := func(b bool) *bool { return &b }
+	controllerRef := &metav1.OwnerReference{
+		APIVersion:         controllerKind.GroupVersion().String(),
+		Kind:               controllerKind.Kind,
+		Name:               rc.Name,
+		UID:                rc.UID,
+		BlockOwnerDeletion: boolPtr(true),
+		Controller:         boolPtr(true),
+	}
+	if pod, err := controller.GetPodFromTemplate(rc.Spec.Template, rc, controllerRef); err == nil {
+		if err = rm.validateNodeConflict(pod); err != nil {
+			return err
+		}
+	}
+
 	return manageReplicasErr
 }
+
+func (rm *ReplicationManager) validateNodeConflict(pod *v1.Pod) error {
+	if len(pod.Spec.NodeName) == 0 {
+		return nil
+	}
+
+	node, err := rm.kubeClient.Core().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
```

Comment:
It's unlike that we are going to do this since I don't think in other failures we need informations from nodes. We basically need a generic mechanism that identifies that the RC is creating the same pods over and over again.